lib/plugins/openaps.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+var _ = require('lodash');
+var moment = require('moment');
+
+var timeago = require('./timeago')();
+
+function init() {
+
+  var openaps = {
+    name: 'openaps'
+    , label: 'OpenAPS'
+    , pluginType: 'pill-status'
+  };
+
+  openaps.setProperties = function setProperties (sbx) {
+    sbx.offerProperty('openaps', function setopenaps ( ) {
+      var openapsStatus = _.findLast(sbx.data.devicestatus, function (status) {
+        return sbx.entryMills(status) <= sbx.time && ('openaps' in status);
+      });
+
+      var loopStatus = checkLoopStatus(openapsStatus, sbx);
+      openapsStatus = openapsStatus || {};
+      openapsStatus.status = loopStatus;
+
+      return openapsStatus;
+    });
+  };
+
+  openaps.updateVisualisation = function updateVisualisation (sbx) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (10)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1432995001/issues/source?bid=2452379&fileBranchId=2989096#l30)